tests/e2e/utils/response_evaluation.py

```diff
@@ -143,3 +151,68 @@ def validate_response(self):
             print("No result. Nothing to process.")
 
         return True
+
+    def evaluate_models(self):
+        """Evaluate models against groundtruth answer."""
+        print("Running model evaluation using groundtruth...")
+        inscope_models = [
+            "bam+ibm/granite-13b-chat-v2",
+            "watsonx+ibm/granite-13b-chat-v2",
+            "openai+gpt-3.5-turbo",
+            "azure_openai+gpt-3.5-turbo",
+        ]
+        answer_id = "ground_truth+with_rag"
+
+        result_df = DataFrame(columns=["eval_id", "question", answer_id])
+        for provider_model in inscope_models:
+            result_dict = defaultdict(list)
+            print(f"Model evaluation for {provider_model}")
+            provider, model = provider_model.split("+")
+            try:
+                prev_resp = read_csv(
+                    f"{self._args.eval_out_dir}/response_evaluation_result-"
+                    f"{provider_model.replace('/', '-')}+with_rag.csv"
+                )
+            except FileNotFoundError:
+                prev_resp = None
+
+            for query_id in self._qa_pairs.keys():
+                question = self._qa_pairs[query_id]["question"]
+                answer_data = self._qa_pairs[query_id]["answer"][answer_id]
+                answer = answer_data["text"][0]
+
+                if prev_resp is not None:
```

Comment:
might be worth to comment what this logic does exactly (at least for me it is not clear). Or better - that might be ideal block to be refactored into it's own function.